src/python/cli_new/lib/cli/http.py

```diff
@@ -64,13 +64,11 @@ def read_endpoint(addr, endpoint, config, query=None):
                            .format(url=url, error=str(exception)))
 
 
-def get_json(addr, endpoint, config, condition=None, query=None):
+def get_json(addr, endpoint, config, query=None):
```

Comment:
So interestingly the only call site which seems to pass 4 arguments  to `get_json` is this: https://github.com/apache/mesos/blob/c66aaf47768aa0e5b66ee9d830f634f4dd682c43/src/python/cli_new/lib/cli/tests/base.py#L280

However it's apparently passing a `timeout` argument, which doesn't seem to be accepted here.
Looking at `git blame`, I think you might have broken it here: https://github.com/apache/mesos/commit/c66aaf47768aa0e5b66ee9d830f634f4dd682c43#diff-1b933ba1cadd6e125668ca2646e6376b740919e25e905142108f9cc324bafb4a

As you can see you removed the `timeout` parameter from `get_json` but the test still uses it.

So I have two questions:
- have you run the tests, or do you know how to run them?
- we should probably fix the test as well

Reply:
Yes I run the test, I will have a look why it's working on my site. Because, u are right, it should fail at that point.

Reply:
ok, after cleanup, the test fails also on my site.  

But the problem is not timeout, it's the removed condition variable. I can re-implement it. But it's only used for the test. :man_shrugging: What do u think? Shell I let it inside or remove it and change the test functions?

Reply:
Well, I decide to change the test functions. If u don't feel fine with it, it's ok and I will re-implement the condition parameter. :-)

Reply:
Thanks, I think that's fine!